### PR TITLE
feat: Redirect story pages to lower case slugs if the slug contains any capital latin letters

### PR DESCRIPTION
**Pull request by EnKrypt**
# Description

We want the feature to redirect story pages to their lower case equivalent if the slug contains any capital letters instead of showing a 404 today. This PR addresses that feature.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Run server locally, pass in a story URL with upper case slug. The server should redirect you to the URL having lowercase slug.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by EnKrypt** [comment]
Tests to be added before merging

**Comment by EnKrypt** [comment]
Could not run auto-formatter as there is no .prettierrc file in the repository.
The local formatter settings on my environment changes the entire file when formatted, so I intentionally saved without formatting.

**Review by sivteck — APPROVED** [review]

File: test/integration/isomorphic-handler-test.js [hunk 1]
```diff
@@ -684,6 +686,73 @@ describe("Isomorphic Handler", function () {
     });
   });
 
+  describe("redirectToLowercaseSlugs", () => {
+    it("when enabled, redirects story slug with capital letters to lowercase slug URL", (done) => {
```
Comment (sharangj) [inline test/integration/isomorphic-handler-test.js:690]: It is enabled

File: test/integration/isomorphic-handler-test.js [hunk 2]
```diff
@@ -684,6 +686,73 @@ describe("Isomorphic Handler", function () {
...
+        .expect("Content-Type", /html/)
+        .expect(200)
+        .then((res) => {
+          const response = JSON.parse(res.text);
+          assert.equal("foobar", response.store.qt.data.text);
+        })
+        .then(done);
+    });
+    
+    it("when not enabled, story slug with capital letters gives a normal response", (done) => {
```
Comment (sharangj) [inline test/integration/isomorphic-handler-test.js:733]: You need to cover a case when it is not story page too. This to protect ourselves from change on code later

File: server/handlers/isomorphic-handler.js [hunk 3]
```diff
@@ -36,8 +36,21 @@ function loadDataForIsomorphicRoute(
 
   // Using async because this for loop reads really really well
   async function loadDataForEachRoute() {
+    const redirectToLowercaseSlugsValue = typeof redirectToLowercaseSlugs === 'function' ? redirectToLowercaseSlugs() : redirectToLowercaseSlugs;
```
Comment (sharangj) [inline server/handlers/isomorphic-handler.js:39]: Do you want to pass the config to the function? This will be helpful.

**Review by sharangj — APPROVED** [review]

**Review by sharangj — CHANGES_REQUESTED** [review]
Sorry approved it by mistake

**Review by sivteck — APPROVED** [review]

**Review by sharangj — APPROVED** [review]
We need to do things before merging. One is a must and the other you can take a call.
1. Use the `config` variable in the `redirectToLowercaseSlugs` function call. (Not a must)
2. Please write an integration guide in developers.quintype.com. You can use the gh-pages branch in the [Malibu](https://github.com/quintype/malibu) github repository.  (This is a must)

Ill approve these changes for now but please dont forget the docs part.

**Pull request merged**